Tests/TestingTests/PollingTests.swift

```diff
@@ -0,0 +1,352 @@
+            var configuration = Configuration()
+            configuration.eventHandler = { event, _ in
+              if case .issueRecorded = event.kind {
+                failed()
+              }
+            }
+            await Test {
+              await #expect(until: .passesAlways) { false }
+            }.run(configuration: configuration)
+          }
+        }
+        #expect(duration.isCloseTo(other: .zero, within: delta))
+      }
+
+      @Test("if the closures starts off as false, but would become true")
+      func changingFromFail() async {
+        let incrementor = Incrementor()
+
+        let duration = await Test.Clock().measure {
+          await confirmation("Polling failed", expectedCount: 1) { failed in
+            var configuration = Configuration()
+            configuration.eventHandler = { event, _ in
+              if case .issueRecorded = event.kind {
+                failed()
+              }
+            }
+            await Test {
+              await #expect(until: .passesAlways) {
+                await incrementor.increment() == 2
+              }
+              // this will pass only on the second invocation
+              // This checks that we fail the test if it immediately returns false
+            }.run(configuration: configuration)
+          }
+        }
+
+        #expect(await incrementor.count == 1)
+        #expect(duration.isCloseTo(other: .zero, within: delta))
+      }
+
+      @Test("if the closure continues to pass")
+      func continuousCalling() async {
+        let incrementor = Incrementor()
+
+        let duration = await Test.Clock().measure {
+          await #expect(until: .passesAlways) {
+            _ = await incrementor.increment()
+            return true
+          }
+        }
+
+        #expect(await incrementor.count > 1)
+        #expect(duration.isCloseTo(other: .seconds(60), within: delta))
+      }
+
+      @Test("Unexpected Errors will automatically exit & fail") func errorsReported() async {
+        let duration = await Test.Clock().measure {
+          await confirmation("Polling failed", expectedCount: 1) { failed in
+            var configuration = Configuration()
+            configuration.eventHandler = { event, _ in
+              if case .issueRecorded = event.kind {
+                failed()
+              }
+            }
+            await Test {
+              await #expect(until: .passesOnce) {
+                throw PollingTestSampleError.ohNo
+              }
+            }.run(configuration: configuration)
+          }
+        }
+        #expect(duration.isCloseTo(other: .zero, within: delta))
+      }
+    }
+  }
+}
+
+private enum PollingTestSampleError: Error {
+  case ohNo
+  case secondCase
+}
+
+extension DurationProtocol {
+  fileprivate func isCloseTo(other: Self, within delta: Self) -> Bool {
+    var distance = self - other
+    if (distance < Self.zero) {
+      distance *= -1
+    }
+    return distance <= delta
+  }
```

Comment:
I want this to be in the swift standard library and apply to all numeric types. It would make comparing floating point types so much more doable.

In my ideal world, this would work like `1.000001 == 1 ± 1e-3` (or `1.00001 == 1 +- 1e-3`, to avoid using the hard-to-type ± symbol). But that would involve adding another ternary operator to the language, and would be much more involved that adding a method to a protocol in the standard library.

Reply:
swift-numerics has this, more or less, but we can't add it as a dependency of course. :)